packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js

```diff
@@ -218,132 +161,160 @@ class MultiSelectTagWidget extends React.Component {
 		);
 	}
 
-	scrollDropDownIfRequired(selectedIndex, direction) {
-		const dropDown = this.component.querySelector('.dropdown-menu');
-		const scrollPosition = dropDown.scrollTop;
-		const dropDownHeight = dropDown.clientHeight;
-		const itemStart = (selectedIndex * DROP_DOWN_ITEM_HEIGHT) + DROP_DOWN_PADDING;
-		const itemEnd = ((selectedIndex + 1) * DROP_DOWN_ITEM_HEIGHT) + DROP_DOWN_PADDING;
+	getDropdownItems(suggestions) {
+		let items = [];
+		if (this.withCategory) {
+			const itemsMap = {};
+			const groupBy = this.props.options.groupBy;
+
+			suggestions.forEach((item) => {
+				itemsMap[item[groupBy]] = itemsMap[item[groupBy]] || [];
+				itemsMap[item[groupBy]].push({ ...item, title: item.label });
+			});
+
+			Object.keys(itemsMap).forEach((category) => {
+				items.push({ title: category, suggestions: itemsMap[category] });
+			});
+		} else if (suggestions && suggestions.length > 0) {
+			items = suggestions.map(item => ({ ...item, title: item.label }));
+		} else {
+			items = suggestions;
+		}
+		return items;
```

Comment:
Why not start by initializing `items` to `suggestions` and then returning it filtered on your if/else statements.